src/simulation.c

```diff
@@ -170,15 +172,24 @@ void simulation_history_to_file(Simulation *sp) {
   } else {
     mkdir(path,0777);
   }
-  sprintf(end, "/%ld", sp->seed);
-  FILE *file = fopen(path,"w");
+
+  char time_path[2048];
+  char* time_end = strcpy(time_path, path);
+  time_end += strlen(time_path);
```

Comment:
this is going to insert random data from the stack into the file path. Line 178 should be deleted.

Also, it would be nice for readability to also have a char `reaction_path[2048]` also for consistency with a few more `strcpy`s. The compiler is smart enough to optimize it out

Reply:
So, I actually found the += strlen(time_path) to be necessary to get the correct paths. If you don't have that line, then the string is overwritten from the beginning, so you end up with something like /times_XXX instead of /base_dir/simulation_histories/times_XXX.

Possible I'm screwing something up elsewhere, but you segfault if you remove this line, and you don't if you keep it.

Reply:
ahhh, OK i see. `strcpy` returns a pointer to `time_path` which is why you need to add the length. I usually use `stpcpy` which returns a pointer to the end